autorest/adal/token.go

```diff
@@ -265,11 +268,18 @@ func NewServicePrincipalTokenFromCertificate(oauthConfig OAuthConfig, clientID s
 }
 
 // NewServicePrincipalTokenFromMSI creates a ServicePrincipalToken via the MSI VM Extension.
-func NewServicePrincipalTokenFromMSI(oauthConfig OAuthConfig, resource string, callbacks ...TokenRefreshCallback) (*ServicePrincipalToken, error) {
-	return newServicePrincipalTokenFromMSI(oauthConfig, resource, managedIdentitySettingsPath, callbacks...)
+func NewServicePrincipalTokenFromMSI(resource string, callbacks ...TokenRefreshCallback) (*ServicePrincipalToken, error) {
```

Comment:
Will it always make sense to fetch the MSI endpoint from a file, or should we be taking the endpoint as one of the parameters here? We could expose convenience functions for fetching the endpoint from a file like you do in the method below.

Reply:
Fixed